src/main/java/com/imaee/propinq/buildings/mappers/BuildingMapper.java

```diff
@@ -0,0 +1,52 @@
+package com.imaee.propinq.buildings.mappers;
+
+import com.imaee.propinq.buildings.controllers.responses.BuildingResponse;
+import com.imaee.propinq.buildings.data.models.Building;
+import com.imaee.propinq.buildings.data.models.BuildingType;
+import com.imaee.propinq.shared.data.models.Review;
+import com.imaee.propinq.shared.data.models.Image;
+import com.imaee.propinq.properties.data.models.Property;
+import com.imaee.propinq.users.data.models.User;
+
+import java.util.List;
+import java.util.UUID;
+
+public class BuildingMapper {
+
+    private BuildingMapper() {}
+
+    public static BuildingResponse toBuildingResponse(Building building) {
+        List<String> imagesURLS = getImagesURLs(building);
+        UUID userId = building.getUser().getUserId();
+        UUID buildingTypeId = building.getBuildingType().getBuildingTypeId();
+        return new BuildingResponse(
+                building.getBuildingId(),
+                building.getName(),
+                building.getDescription(),
+                building.getAddress(),
+                imagesURLS,
+                userId,
+                buildingTypeId
+        );
+    }
+    private static List<String> getImagesURLs(Building building) {
+        return building.getImages().stream()
+                .map(image -> image.getUrl())
+                .toList();
+    }
```

Comment:
Este método está pegado a la llave de cierre del método anterior. Ojo con estos detalles.

Respecto a la implementación: (1) este método debería ser un método privado en el `BuildingService` y (2) para tirar un poco más de facha (y más importante, delegarle el control a la API de streams), podés pasarle la referencia de la función al `map` en vez de crear la lambda de forma explícita

```suggestion
    private static List<String> getImagesURLs(Building building) {
        return building.getImages().stream()
                .map(Image::getUrl)
                .toList();
    }
```

A esto último me vas a ver hacerlo bastante seguido jeje

Reply:
Lo de no utilizar la lambda, solo aplica para cuando dentro del .map le pasamos un solo argumento, en cambio si queremos pasar más de un argumento al método, no hay de otra opción verdad?